core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -54,22 +56,24 @@
   private final TableOperations ops;
   private final Table table;
   private final Long snapshotId;
+  private final Map<String, String> options;
   private final Schema schema;
   private final Expression rowFilter;
   private final boolean caseSensitive;
   private final boolean colStats;
   private final Collection<String> selectedColumns;
 
   protected BaseTableScan(TableOperations ops, Table table, Schema schema) {
-    this(ops, table, null, schema, Expressions.alwaysTrue(), true, false, null);
+    this(ops, table, null, null, schema, Expressions.alwaysTrue(), true, false, null);
   }
 
-  protected BaseTableScan(TableOperations ops, Table table, Long snapshotId, Schema schema,
-                        Expression rowFilter, boolean caseSensitive, boolean colStats,
+  protected BaseTableScan(TableOperations ops, Table table, Long snapshotId, Map<String, String> options,
```

Comment:
Why add options in the middle instead of at the end? Seems like you're changing more lines by displacing the other arguments.

Reply:
Fixed to be added to the end.

The reason it was in the middle before is that it used to be `splitOptions` and those were related to `snapshotId` as in that they modify the read behavior.

Now they're generic so agreed it makes sense to keep at end.